melange development environment

puerco opened the pull request:
This PR introduces the first iteration of the melange development environment. It is based on the apko equivalent
with two main differences:

1. It uses our apko image to build the devenv image
2. It is based on Wolfi packages :tada:

To run it, execute the same command as the apko devenv:

./hack/make-devenv.sh

It requires https://github.com/wolfi-dev/os/pull/118 to fix a bug with the wolfi key locations and to enable https://github.com/wolfi-dev/os/pull/119 which is required to build the `alpine-keys` wolfi package (required to build the devenv image).


Signed-off-by: Adolfo García Veytia (Puerco) <puerco@chainguard.dev>

jdolitsky commented on hack/make-devenv.sh:
```diff
@@ -0,0 +1,74 @@
+
+IMAGE_TAG="melange-inception"
+DEVENV_IMAGE_TARBALL="melange-inception.tar.gz"
+BUILDER_APKO_TAG="sha256:c96d1e6886ae5bafafe2656a3133ee73ce5245f67e60e430a731570b3f76797e"
```
wondering if latest is more appropriate, since this will get out of date in < 24 hrs

puerco replied:
Updated to latest

jdolitsky commented on hack/make-devenv.sh:
```diff
@@ -0,0 +1,74 @@
+}
+
+run() {
+    docker run --rm --privileged -w /melage -v /var/run/docker.sock:/var/run/docker.sock \
+        -v $(pwd):/melage -ti ${IMAGE_TAG}:latest hack/make-devenv.sh setup
```
s/melage/melange/g

puerco replied:
Thanks!

jdolitsky reviewed (changes requested):
overall lgtm!

This does have a bit of overlap w/ the SDK image: https://github.com/chainguard-images/sdk/blob/main/USAGE.md

puerco commented:
> This does have a bit of overlap w/ the SDK image: https://github.com/chainguard-images/sdk/blob/main/USAGE.md

He, sorry @jdolitsky I didn't remember the sdk :upside_down_face: 

Until now I had been running melange using this script adapted from the apko hack I wrote a while back, I am happy to drop this (and the apko devenv script) in favor of the SDK and focus on that one.

puerco commented:
Now that we have the alpine-keys package, this should be ready to merge, I've rebased this and addressed the comments @jdolitsky , feel free to close it if it doesn't make sense to keep this and the SDK image :)
